sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -860,9 +872,10 @@ export class ServiceBusAtomManagementClient extends ServiceClient {
       response.parsedBody == undefined ||
       (Array.isArray(response.parsedBody) && response.parsedBody.length == 0)
     ) {
+      response.status = 404;
       const err = new RestError(
```

Comment:
Can we not move the case of creating message entity not found error in case of `NoContent` into the `buildError()` utility function? That way all error related processing can be made at a single place.

Reply:
In general, there always more than one way of doing things. At this point, I don't think goal is to have code be exactly same as .NET and we should likely focus on high level use-cases?

The 204 being a non-successful response is a workaround that we are having to add. 
https://docs.microsoft.com/en-us/uwp/api/windows.web.http.httpresponsemessage.issuccessstatuscode Acc. to this 204 is still a successful code. Perhaps .NET SDK has some workaround to exclude that in their pipeline. (Our pipeline implementations are different, so that's another reason why we should only be looking at high levelscenarios I think.)

Another reason is 204 may not always be a non-successful status, we know that only for the Get API we needed to make this exception.

Reply:
We discussed in meeting with @bterlson as well that the goal wasn't to replicate .NET SDK line by line.

Reply:
If we do not want to or cannot rely on the status code 204 of NoContent to determine the case of Message Entity Not Found and want to treat empty response body or empty array in response body as a 404 directly this way, then I would suggest using the error code directly here instead of calling the `getErrorCode(response)`.

At this point of the code path, we are aware of what error code and status code to use and so can do the same directly. Calling `getErrorCode()` helps when the response content and status needs to be looked into to determine error code.

I would also suggest not updating the `response.status` and have it represent what ever is the status returned by the service.

Reply:
Updated